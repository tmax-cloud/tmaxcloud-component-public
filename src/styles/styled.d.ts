import "styled-components";

declare module "styled-components" {
  export interface DefaultTheme {
    font: any;
    color: any;
  }
}
