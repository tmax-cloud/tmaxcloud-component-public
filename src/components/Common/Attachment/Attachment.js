import PropTypes from "prop-types";
import styled, { css } from "styled-components";

const Wrapper = styled.div`
  width: fit-content;
  display: flex;
  flex-direction: row;
  align-items: ${({ state }) => (state === "Medium" ? "center" : "flex-start")};
  justify-content: center;
  gap: 0.8rem;
  background: ${({ theme }) => theme.color.white._100};
  border: 0.1rem solid ${({ theme }) => theme.color.gray._300};
  border-radius: 0.8rem;
  padding: 1.25rem 1.2rem;
`;

const IconWrapper = styled.div``;

const Icon = styled.img`
  width: 2rem;
  height: 2rem;
`;

const DataWrapper = styled.div`
  display: flex;

  ${({ state }) => {
    if (state === "Medium")
      return css`
        flex-direction: row;
        gap: 0.8rem;
        align-items: center;
      `;
    else
      return css`
        flex-direction: column;
        gap: 0.3rem;
        align-items: flex-start;
      `;
  }}
`;
const Name = styled.div`
  ${({ theme }) => theme.font.body2_400};
  color: ${({ theme }) => theme.color.gray._900};
`;

const Size = styled.div`
  ${({ theme }) => theme.font.body3_400};
  color: ${({ theme }) => theme.color.gray._500};
`;

const Attacthment = ({ state, name, size }) => {
  return (
    <Wrapper state={state}>
      <IconWrapper>
        <Icon src="/" alt="file" />
      </IconWrapper>
      <DataWrapper state={state}>
        <Name>{name}</Name>
        <Size>{size}</Size>
      </DataWrapper>
    </Wrapper>
  );
};

Attacthment.propTypes = {
  /** 타입 */
  state: PropTypes.oneOf(["Medium", "Large"]),
  /** 파일 이름 */
  name: PropTypes.string,
  /** 파일 크기 */
  size: PropTypes.string,
};

Attacthment.defaultProps = {
  state: "Medium",
  name: "이름",
  size: "사이즈",
};

export default Attacthment;
