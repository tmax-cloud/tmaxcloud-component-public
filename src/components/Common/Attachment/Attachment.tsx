import styled, { css } from "styled-components";
import React from "react";
/** Attachment Props 타입 */
type AttachmentPropsType = {
  /** 타입 */
  state: "Medium" | "Large";
  /** 파일 이름 */
  name: string;
  /** 파일 크기 */
  size: string;
};

const Wrapper = styled.button`
  width: fit-content;
  display: flex;
  flex-direction: row;
  align-items: ${({ state }) => (state === "Medium" ? "center" : "flex-start")};
  justify-content: center;
  gap: 0.8rem;
  background: ${({ theme }) => theme.color.white._100};
  border: 0.1rem solid ${({ theme }) => theme.color.gray._300};
  border-radius: 0.8rem;
  padding: 1.25rem 1.2rem;
  :hover {
    background-color: ${({ theme }) => theme.color.gray._50};
    border: 0.1rem solid ${({ theme }) => theme.color.gray._500};
  }
  :active,
  :focus-visible,
  :focus {
    background-color: ${({ theme }) => theme.color.gray._100};
    border: 0.1rem solid ${({ theme }) => theme.color.gray._900};
  }
`;

const IconWrapper = styled.div``;

const Icon = styled.img`
  width: 2rem;
  height: 2rem;
`;

const DataWrapper = styled.div`
  display: flex;

  ${({ state }) => {
    if (state === "Medium")
      return css`
        flex-direction: row;
        gap: 0.8rem;
        align-items: center;
      `;
    else
      return css`
        flex-direction: column;
        gap: 0.3rem;
        align-items: flex-start;
      `;
  }}
`;
const Name = styled.div`
  ${({ theme }) => theme.font.body2_400};
  color: ${({ theme }) => theme.color.gray._900};
`;

const Size = styled.div`
  ${({ theme }) => theme.font.body3_400};
  color: ${({ theme }) => theme.color.gray._500};
`;
/** Attachment */
const Attachment = ({ state, name, size }: AttachmentPropsType) => {
  return (
    <Wrapper state={state}>
      <IconWrapper>
        <Icon src="/asset/images/Icon/dummy_icon.svg" alt="file" />
      </IconWrapper>
      <DataWrapper state={state}>
        <Name>{name}</Name>
        <Size>{size}</Size>
      </DataWrapper>
    </Wrapper>
  );
};

Attachment.defaultProps = {
  state: "Medium",
  name: "이름",
  size: "사이즈",
};

export default Attachment;
