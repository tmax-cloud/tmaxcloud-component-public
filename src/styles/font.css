@font-face {
  font-family: "SpoqaHanSansNeo-Bold";
  src: url("./fonts/SpoqaHanSansNeo-Bold.eot") format("embedded-opentype"), url("./fonts/SpoqaHanSansNeo-Bold.woff2") format("woff2");
  font-display: swap;
}
@font-face {
  font-family: "SpoqaHanSansNeo-Medium";
  src: url("./fonts/SpoqaHanSansNeo-Medium.eot") format("embedded-opentype"), url("./fonts/SpoqaHanSansNeo-Medium.woff2") format("woff2");
  font-display: swap;
}
@font-face {
  font-family: "SpoqaHanSansNeo-Regular";
  src: url("./fonts/SpoqaHanSansNeo-Regular.eot") format("embedded-opentype"), url("./fonts/SpoqaHanSansNeo-Regular.woff2") format("woff2");
  font-display: swap;
}
@font-face {
  font-family: "SpoqaHanSansNeo-Light";
  src: url("./fonts/SpoqaHanSansNeo-Light.eot") format("embedded-opentype"), url("./fonts/SpoqaHanSansNeo-Light.woff2") format("woff2");
  font-display: swap;
}

@font-face {
  font-family: "SpoqaHanSansNeo-Thin";
  src: url("./fonts/SpoqaHanSansNeo-Thin.eot") format("embedded-opentype"), url("./fonts/SpoqaHanSansNeo-Thin.woff2") format("woff2");
  font-display: swap;
}
