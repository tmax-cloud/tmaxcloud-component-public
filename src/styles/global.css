@import url(font.css);
html {
  font-size: 62.5%;
}
body {
  padding: 0;
  margin: 0;
  word-break: keep-all;
}

a {
  color: inherit;
  text-decoration: none;
  background-color: transparent;
  outline: none;
  cursor: pointer;
  transition: color 0.3s;
}
a:hover,
a:active {
  color: inherit;
}

ul,
li,
h2,
h3,
button {
  all: unset;
  box-sizing: border-box;
  margin: 0;
  padding-inline-start: 0;
  padding-inline-end: 0;
  margin-block: 0;
  margin-inline-start: 0;
  margin-inline-end: 0;
}
ul,
li {
  all: unset;
  padding-inline-start: 0;
  padding-inline-end: 0;
  margin-block: 0;
  margin-inline-start: 0;
  margin-inline-end: 0;
}
h1,
h2,
h3 {
  display: block;
}
* {
  box-sizing: border-box;
}
/* Notification */
@keyframes FadeIn {
  from {
    transform: translate(-50%, 3rem);
    opacity: 0;
  }

  to {
    transform: translate(-50%, 0);
    opacity: 1;
  }
}
@keyframes FadeOut {
  from {
    transform: translate(-50%, 0);
    opacity: 1;
  }

  to {
    transform: translate(-50%, 3rem);
    opacity: 0;
    visibility: hidden;
  }
}
@keyframes RadioBtnClickScale {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(0.5);
  }
}
@keyframes RadioBtnClickGrow {
  0% {
    box-shadow: 0 0 0 0.1rem rgba(24, 144, 255, 0.12);
  }
  100% {
    box-shadow: 0 0 0 0.3rem rgba(24, 144, 255, 0.12);
  }
}
@keyframes RadioBtnUnClick {
  0% {
    background-color: #3864ff;
  }
  100% {
    background-color: #ffffff;
  }
}
