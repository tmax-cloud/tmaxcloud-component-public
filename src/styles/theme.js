const font = {
  header1_700: `
    font-family:'SpoqaHanSansNeo-Bold';
    font-size:3.2rem;
    font-weight:700;
    `,
  header2_700: `
    font-family:'SpoqaHanSansNeo-Bold';
    font-size:2.8rem;
    font-weight:700;
    `,
  header3_700: `
    font-family:'SpoqaHanSansNeo-Bold';
    font-size:2.2rem;
    font-weight:700;
    `,
  header3_300: `
    font-family:'SpoqaHanSansNeo-light';
    font-size:2.2rem;
    font-weight:300;
    `,
  header4_500: `
    font-family:'SpoqaHanSansNeo-Medium';
    font-size:2.0rem;
    font-weight:500;
    `,
  header5_700: `
    font-family:'SpoqaHanSansNeo-Bold';
    font-size:1.8rem;
    font-weight:700;
    `,
  header5_500: `
    font-family:'SpoqaHanSansNeo-Medium';
    font-size:1.8rem;
    font-weight:500;
    `,
  header5_400: `
    font-family:'SpoqaHanSansNeo-Regular';
    font-size:1.8rem;
    font-weight:400;
    `,
  body1_700: `
    font-family:'SpoqaHanSansNeo-Bold';
    font-size:1.6rem;
    font-weight:700;
    `,
  body1_400: `
    font-family:'SpoqaHanSansNeo-Regular';
    font-size:1.6rem;
    font-weight:400;
    `,
  body2_700: `
    font-family:'SpoqaHanSansNeo-Bold';
    font-size:1.4rem;
    font-weight:700;
    `,
  body2_500: `
    font-family:'SpoqaHanSansNeo-Medium';
    font-size:1.4rem;
    font-weight:500;
    `,
  body2_400: `
    font-family:'SpoqaHanSansNeo-Regular';
    font-size:1.4rem;
    font-weight:400;
    `,
  body3_700: `
    font-family:'SpoqaHanSansNeo-Bold';
    font-size:1.3rem;
    font-weight:700;
    `,
  body3_500: `
    font-family:'SpoqaHanSansNeo-Medium';
    font-size:1.3rem;
    font-weight:500;
    `,
  body3_400: `
    font-family:'SpoqaHanSansNeo-Regular';
    font-size:1.3rem;
    font-weight:400;
    `,
  body4_700: `
    font-family:'SpoqaHanSansNeo-Bold';
    font-size:1.2rem;
    font-weight:700;
    `,
  body4_400: `
    font-family:'SpoqaHanSansNeo-Regular';
    font-size:1.2rem;
    font-weight:400;
    `,
  detail1_700: `
    font-family:'SpoqaHanSansNeo-Bold';
    font-size:1.1rem;
    font-weight:700;
    `,
  detail1_500: `
    font-family:'SpoqaHanSansNeo-Medium';
    font-size:1.1rem;
    font-weight:500;
    `,
  detail1_400: `
    font-family:'SpoqaHanSansNeo-Regular';
    font-size:1.1rem;
    font-weight:400;
    `,
  detail2_400: `
    font-family:'SpoqaHanSansNeo-Regular';
    font-size:1.0rem;
    font-weight:400;
    `,
  detail3_400: `
    font-family:'SpoqaHanSansNeo-Regular';
    font-size:.8rem;
    font-weight:400;
    `,
};
const color = {
  marine: {
    _700: `
    #0D2BC9;
  `,
    _600: `
  #1C3DE8;
  `,
    _500: `
  #3864FF;
  `,
    _400: `
  #518CFF;
  `,
    _300: `
  #8EB5FF;
  `,
    _200: `
  #B1D0FF;
  `,
    _100: `
  #D3E5FF;
  `,
    _50: `
  #E4EFFF;
  `,
  },
  navy: {
    _700: `
    #0D1B32;
  `,
    _600: `
    #001940;
  `,
    _500: `
    #07224B;
  `,
    _400: `
    #1F375B;
  `,
    _300: `
    #556C8D;
  `,
    _200: `
    #758AAA;
  `,
    _100: `
    #ADBCD6;
  `,
    _50: `
    #CDD6E5;
`,
  },
  violet: {
    _700: `
  #3C13B2;
  `,
    _600: `
      #5100D6;
  `,
    _500: `
      #6A37FF;
  `,
    _400: `
      #6850FF;
  `,
    _300: `
      #8378FF;
  `,
    _200: `
      #9B99FF;
  `,
    _100: `
      #ADABF3;
  `,
    _50: `
      #D2D1FF;
  `,
  },
  sky: {
    _700: `
      #006DD2;
  `,
    _600: `
      #009AF1;
  `,
    _500: `
      #09B5FF;
  `,
    _400: `
      #1AC8FF;
  `,
    _300: `
      #20D7FF;
  `,
    _200: `
      #7AE7FF;
  `,
    _100: `
      #B2F1FF;
  `,
    _50: `
      #CAF5FF;
  `,
  },
  peacock: {
    _700: `
      #00426B;
  `,
    _600: `
      #125D92;
  `,
    _500: `
      #0099CA;
  `,
    _400: `
      #0DB2D7;
  `,
    _300: `
      #42D2F2;
  `,
    _200: `
      #7BE7FF;
  `,
    _100: `
      #BCF3FF;
  `,
    _50: `
      #D7F8FF;
  `,
  },
  emerald: {
    _700: `
      #00616F;
  `,
    _600: `
      #008E8E;
  `,
    _500: `
      #03BEB6;
  `,
    _400: `
      #00CABE;
  `,
    _300: `
      #5AE3DB;
  `,
    _200: `
      #9BEAE5;
  `,
    _100: `
      #C7F1EF;
  `,
    _50: `
      #E4F6F3;
  `,
  },
  pink: {
    _700: `
      #BF3D5C;
  `,
    _600: `
      #E24F6A;
  `,
    _500: `
      #FF5775;
  `,
    _400: `
      #FF6B86;
  `,
    _300: `
      #FF8392
  `,
    _200: `
      #FFADB7;
  `,
    _100: `
      #FFCAD3;
  `,
    _50: `
      #FFE4E6;
  `,
  },
  tangerine: {
    _700: `
      #FF531D;
  `,
    _600: `
      #FF6C2E;
  `,
    _500: `
      #FF814B;
  `,
    _400: `
      #FF814B;
  `,
    _300: `
      #FFA756;
  `,
    _200: `
      #FFC876;
  `,
    _100: `
      #FFDDA3;
  `,
    _50: `
      #FFE8CF;
  `,
  },
  mustard: {
    _700: `
      #FFB700;
  `,
    _600: `
      #FFC300;
  `,
    _500: `
      #FFD600;
  `,
    _400: `
      #FEDF6A;
  `,
    _300: `
      #FFE78D;
  `,
    _200: `
      #FFF0A4;
  `,
    _100: `
      #FFF9BF;
  `,
    _50: `
      #FFFDD9;
  `,
  },
  black: {
    _100: `
    rgba(0, 0, 0, 1);
  `,
    _80: `
    rgba(0, 0, 0, 0.8);
  `,
    _70: `
    rgba(0, 0, 0, 0.7);
  `,
    _50: `
    rgba(0, 0, 0, 0.5);
  `,
    _40: `
    rgba(0, 0, 0, 0.4);
  `,
    _30: `
    rgba(0, 0, 0, 0.3);
  `,
    _20: `
    rgba(0, 0, 0, 0.2);
  `,
    _10: `
    rgba(0, 0, 0, 0.1);
  `,
    _6: `
  rgba(0, 0, 0, 0.06);
  `,
    _4: `
  rgba(0, 0, 0, 0.04);
  `,
  },
  white: {
    _100: `
    rgba(255, 255, 255, 1);
  `,
    _90: `
    rgba(255, 255, 255, 0.9);
  `,
    _80: `
    rgba(255, 255, 255, 0.8);
  `,
    _70: `
    rgba(255, 255, 255, 0.7);
  `,
    _50: `
    rgba(255, 255, 255, 0.5);
  `,
    _40: `
    rgba(255, 255, 255, 0.4);
  `,
    _20: `
    rgba(255, 255, 255, 0.2);
  `,
    _0: `
    rgba(0, 0, 0, 1);
  `,
  },
  gray: {
    _900: `
    #222222;
    `,
    _800: `
    #424242;
    `,
    _700: `
    #616161;
  `,
    _600: `
    #8A8A8A;
  `,
    _500: `
    #9E9E9E;
  `,
    _400: `
    #BDBDBD;
  `,
    _300: `
    #E0E0E0;
  `,
    _200: `
    #EEEEEE;
  `,
    _150: `
  #F2F2F2;
  `,
    _100: `
    #F5F5F5;
  `,
    _50: `
    #FAFAFA;
  `,
  },
  blueGray: {
    _900: `
    #1D1E1F;
    `,
    _800: `
    #292E35;
    `,
    _700: `
    #3D4551;
  `,
    _600: `
    #48515B;
  `,
    _500: `
    #5F6774;
  `,
    _400: `
    #9199A5;
  `,
    _300: `
    #AFB3BD;
  `,
    _200: `
    #C4C8D0;
  `,
    _150: `
  #D9DCE3;
  `,
    _100: `
    #EBECEF;
  `,
    _50: `
    #F5F5F5;
  `,
  },
  info: {
    _100: `#0F8CFF`,
    _font: `#007AEC`,
    _fill: `#F5FAFF`,
  },
  success: {
    _100: `#00D97E`,
    _font: `#00BA77`,
    _fill: `#E6FBF2`,
  },
  error: {
    _100: `#F44336`,
    _font: `#F81B0A`,
    _fill: `#FEECEB`,
  },
  warning: {
    _100: `#FF981C`,
    _font: `#FF7A00`,
    _fill: `#FFEFDB`,
  },
  notice: {
    _100: `#FAD749`,
    _font: `#F6C500`,
    _fill: `#FEF7DB`,
  },

  template: {
    _700: `
   
  `,
    _600: `
  
  `,
    _500: `
    
  `,
    _400: `
    
  `,
    _300: `
    
  `,
    _200: `
    
  `,
    _100: `
    
  `,
    _50: `
    
  `,
  },
};

export const defaultTheme = {
  font,
  color,
};
