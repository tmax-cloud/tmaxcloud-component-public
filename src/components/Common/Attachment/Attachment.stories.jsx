import Attachment from "./Attachment";

export default {
  title: "common/Attachment",
  component: Attachment,
  argTypes: {},
  decorators: [],
  parameters: {
    componentSubtitle: "첨부파일의 확장자 및 정보를 나타내는 컴포넌트.",
  },
};

const Template = (args) => <Attachment {...args} />;

export const AttachmentTemplate = Template.bind({});

AttachmentTemplate.args = {
  state: "Large",
  name: "Attachment.png",
  size: "80KB",
};
