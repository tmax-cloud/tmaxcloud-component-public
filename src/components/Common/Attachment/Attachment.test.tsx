import Attachment from "./Attachment";
import { render, screen } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { defaultTheme } from "styles/theme";
describe("Attachment 공통 테스트", () => {
  it("matches snapshot", () => {
    const utils = render(
      <ThemeProvider theme={defaultTheme}>
        <Attachment state="Medium" name="이름" size="크기" />
      </ThemeProvider>,
    );
    expect(utils.container).toMatchSnapshot();
  });
  it("텍스트 노출 확인", () => {
    render(
      <ThemeProvider theme={defaultTheme}>
        <Attachment state="Medium" name="이름" size="크기" />
      </ThemeProvider>,
    );
    expect(screen.getByText("이름")).toBeTruthy();
    expect(screen.getByText("크기")).toBeTruthy();
    console.log(screen.getByText("크기").style.color);
  });
});
