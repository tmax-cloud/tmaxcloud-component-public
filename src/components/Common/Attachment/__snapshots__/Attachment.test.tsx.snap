// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attachment 공통 테스트 matches snapshot 1`] = `
<div>
  <button
    class="sc-bczRLJ ecuuZT"
  >
    <div
      class="sc-gsnTZi"
    >
      <svg
        class="sc-dkzDqf gfccow"
      >
        dummy_icon.svg
      </svg>
    </div>
    <div
      class="sc-hKMtZM elcAqk"
    >
      <div
        class="sc-eCYdqJ brbXSe"
      >
        이름
      </div>
      <div
        class="sc-jSMfEi bhbPBC"
      >
        크기
      </div>
    </div>
  </button>
</div>
`;
